### DISCUSSION TITLE
Nested todo

### DISCUSSION THREAD

**Discussion by citiz3n404** [ransome1/sleek#467] [Ideas]
I tested sleek for a week and I think It would be great If we could have some subtasks. If all the subtasks are completed then the task is automatically checked.

- [ ] TASK1
- [ ] TASK2
        - [ ] SUBTASK1
        - [ ] SUBTASK2

**Comment by warmwaffles** [comment]
I don't think the Todo.txt spec allows for that support.

**Reply by citiz3n404** [reply]
We can imagine a system with an ID attributed to each task. 
For the Todo.txt, it will not be native, of course, but we can do that by starting the Description part of the note by #ID_task #ID_master_task.

Exemple :
`x (A) 2023-02-23 2023-03-23 #ID_task  #ID_master_task Do my homework +project1 @context due:2023-03-23`

The description part would be `#ID_task  #ID_master_task Do my homework` and sleek would just not print the IDs.

Of course this part would be custom and if we want to stick to the specs, I guess it will not be done because the usage would be specific to sleek.

**Comment by CPUGPU** [comment]
Subtasks cannot be done within [the rules of todo.txt](https://github.com/todotxt/todo.txt).

However, there are other ways to group and sort tasks within the capabilities of todo.txt.  I use separate files for significant projects, which is where Sleek's multiple-tab interface really shines.  I also use projects tags for... well... projects!  Even contexts can help group tasks based on where they need to get done.

TL;DR - There are ways to manage project subtasks already in the todo.txt spec, and I think Sleek should stick to that spec.

**Reply by citiz3n404** [reply]
Yeah the idea to have a tab per project is really good and it's what I'm currently doing but still, inside a project sometime I would love to nest some tasks. 
I would understand If sleek only want to be a todo.txt viewer.

**Reply by elijaheyes** [reply]
What's the reason it can't be done?

I think as long as the task ID doesn't contain whitespace characters or colons, then it should work within the rules laid out for extra metadata (additional tags). For example:

`x (A) 2023-03-23 2023-02-23 do all homework +project @context due:2023-03-23 uuid:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`
`x (A) 2023-03-22 3034-02-23 do math homework +project @context due:2023-03-23 uuid:yyyyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy parent:xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx`

math homework is subtask to all homework

**Comment by elijaheyes** [comment]
In leiu of subtask functionality, I've found that if you add multiple contexts to a task, sleek will sort based on the first context, then by the second context, etc. I'm currently using this to organize tasks into sub-groups. I'm sure the same can be done for projects.

-----

I had a similar idea for ID_task and ID_master_task.
Perhaps the ID could be a uuid, and the new tags could be "uuid:" and "parent:"
ID_task: and ID_master_task: tags also makes sense.

-----

What is the reason this wouldn't fit into the Todo.txt spec?

Just looking to understand how to utilize todotxt as it was intended to be used.
Is there an advantage for not allowing sub-tasks?

I don't see anything written in the format rules saying that subtasks aren't allowed.
On the contrary, it seems todotxt format encourages the use of subtasks by including a project tag within the format.
However, users like myself find project and context tags limiting because it only allows for one word per project or context.
It would be really great to find a way to use the existing format for true subtask support (projects contain tasks, not sub-tasks).
And if it doesn't, to find a way to extend the format with additional tags like what has already been done for other features.

-----

Is there a way to write lengthier project text?
That way a project can be written the same way as a task.

-----

Another idea I had was allowing drag&drop functionality.
Drag task B on top of another task A and have sleek automatically add task B as a subtask to task A.
Then the subtasks can be viewed with a drop-down arrow on the right of the task.

**Comment by new-creation7** [comment]
This is what '+Project' is for.
